rep-0003.rst

```diff
@@ -337,6 +337,8 @@ Requirements:
 
 " ! " means that this package will be at least this version (since these distributions have not yet been released); this may change as those releases get closer.
 
+Note that Gazebo is a special case, since it spans versions 7, 8, and 9.  The only long-term supported platform (Bionic) has Gazebo 9, so packages should target that.  Packages are encouraged to additionally support Gazebo 7, but it is not required.
```

Comment:
Maybe it's a matter of wording but Debian Stretch is LTS and supported, isn't it ?
Or has the decision of shipping Gazebo 9 on Debian Stretch been made ?

Reply:
I think it is a matter of wording.  Debian Stretch is only "recommended" support, not "required" support for Melodic.  I'll change the wording to say that Bionic is the only long-term supported required platform, or something to that effect.

Reply:
Take a look at the wording in be648e8 and let me know if it is clearer.

Reply:
> Packages are encouraged to additionally support Gazebo 7, but it is not required.

Gazebo 7 and 8 based on the previous sentence as some "recommended platforms" ship with Gazebo 8 at the moment.

I still think we should confirm where we stand on Debian Stretch (use 7 from upstream or distribute 9 in the ROS apt repo) before considering merging this

Reply:
I've updated the language to say "7 and 8".  After talking to @j-rivero , he is pretty confident that we can get Gazebo 9 packages built for Debian Stretch.  So I've updated the document to have Debian Stretch be 9.0.0, with a star to indicate it is an OSRF package.

Reply:
Is gazebo 9.0.0 side-by-side installable with the version provided upstream (7.3.1)? if not that means that Lunar and Melodic will not be side-by-side installable on Debian Stretch

Reply:
Mikael is right, using gazebo9 we won't be able to provide the Debian Stretch users with the option of having Lunar and Melodic desktop-full (or supporting Gazebo at the same time). We have discussed this before but since there was not collision in Ubuntu, we did not find a compromise for the future.

I'm afraid that the option of using `gazebo7` for Debian in `Stretch`will make the things more difficult since there is two API bumps between versions in Ubuntu/Debian Stretch. I don't know how hard it is the limitation of not having two ROS versions on Debain Stretch but seems to me that i could be better than having to deal with two different versions of Gazebo in ROS Melodic.

Reply:
> using gazebo7 [..] will make the things more difficult since there is two API bumps between versions in Ubuntu/Debian Stretch.

Agreed, we discussed it recently and as we didn't know in details how big the API breakage was we didn't settle on it. Note that this issue (of supporting both Gazebo 7 and 9 APIs in Melodic) also exists with Artful that is a platform for which we "require support".

If the consensus is that it'll be too much of a burden for maintainers to support both APIs I suggest the following:
- Shipping Gazebo 9 on Debian Stretch and add a note to tell user that this ROS version is not side by side installable with the previous one (not sure what the right place for this is but I think it should be made pretty visible as this never happened before AFAIK)
- Move Artful to the "recommended support" platforms and create separate build files for it to allow people to blacklist it if they don't intend to support it.
- If we keep Artful, (even only in recommended support), we would need at least `desktop-full` on all platforms we build debs for. Do you think it's doable to have a version of `gazebo_ros_pkgs` in melodic that will support both APIs?
  - If that's not the case we may want either:
    - Reconsider Artful support as a whole for Melodic
    - Consider shiping Gazebo 9 on Artful as well

If the API breakage is not significant to justify diverging from upstream:
- Use upstream on Stretch
- Use upstream on Artful and keep it in the "required support" section

Reply:
> Do you think it's doable to have a version of gazebo_ros_pkgs in melodic that will support both APIs?

The current `lunar-devel` branch supports gazebo7, gazebo8 and gazebo9 and it will be the base of `melodic-devel` so yes, we can keep the support if we have a reason for that.

> If the API breakage is not significant to justify diverging from upstream:

The API changes between 7.x and 9.x are many:
https://bitbucket.org/osrf/gazebo/src/default/Migration.md?fileviewer=file-view-default

Reply:
In our offline conversations, @j-rivero agreed that he would build Gazebo 9 packages for Ubuntu Artful and Debian Stretch.  Thus, I have updated this PR to call for Gazebo 9 across the board (with the exception of Fedora, which is only recommended anyway).  I also did a small update to the package versions in Bionic and Fedora.  I think this should be ready to review now.